tracer/src/Datadog.Trace/OTelMetrics/MetricReader.cs

```diff
@@ -8,78 +8,131 @@
 #nullable enable
 
 using System;
+using System.Diagnostics.Metrics;
 using System.Threading;
+using System.Threading.Tasks;
+using Datadog.Trace.Configuration;
 using Datadog.Trace.Logging;
 
-namespace Datadog.Trace.OTelMetrics;
-
-internal static class MetricReader
+namespace Datadog.Trace.OTelMetrics
 {
-    private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(MetricReader));
-
-    private static System.Diagnostics.Metrics.MeterListener? _meterListenerInstance;
-    private static int _initialized;
-    private static int _stopped;
-
-    public static bool IsRunning =>
-        Interlocked.CompareExchange(ref _initialized, 1, 1) == 1 &&
-        Interlocked.CompareExchange(ref _stopped, 0, 0) == 0;
-
-    public static void Initialize()
+    /// <summary>
+    /// Instance-based metric reader that collects and exports metrics on a periodic interval.
+    /// Owns the MeterListener and coordinates between the handler and exporter.
+    /// </summary>
+    internal sealed class MetricReader
     {
-        if (Interlocked.CompareExchange(ref _initialized, 1, 0) == 1)
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(MetricReader));
+        private readonly TracerSettings _settings;
+        private readonly MetricReaderHandler _handler;
+        private readonly MetricExporter _exporter;
+        private MeterListener? _listener;
+        private Timer? _timer;
+
+        public MetricReader(TracerSettings settings, MetricReaderHandler handler, MetricExporter exporter)
         {
-            return;
+            _settings = settings;
```

Comment:
Rather than keep a reference to the settings object, can you copy over the values you need as fields?